src/components/virtualRepeat/virtual-repeater.js

```diff
@@ -501,6 +501,12 @@ function VirtualRepeatController($scope, $element, $attrs, $browser, $document,
   this.blocks = {};
   /** @type {Array<!VirtualRepeatController.Block>} A pool of presently unused blocks. */
   this.pooledBlocks = [];
+
+  $scope.$on('$destroy', angular.bind(this, function cleanup() {
```

Comment:
Definitely needs a unit test to make ensure that this doesn't regress in the future.